sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ParseMode.scala

```diff
@@ -51,6 +56,8 @@ object ParseMode extends Logging {
     case PermissiveMode.name => PermissiveMode
     case DropMalformedMode.name => DropMalformedMode
     case FailFastMode.name => FailFastMode
+    case NullMalformedMode.name => NullMalformedMode
```

Comment:
I think we should only add the new mode for `from_json`, not the entire json data source?

Reply:
I would add it to `from_json` but the function shares the same common classes as JSON data source, and the mode is used in the common classes. Any ideas how to restrict the mode value to `from_json` only?

Reply:
After looking at it again, the `mode` option is already there, but `from_json` ignores it.

Now this PR looks like a bug fix to me. I'm wondering if we do need to keep the previous behavior. But if we do, I think using a SQL config is more reasonable than adding a new mode.

Reply:
ok. I will revert the last changes back.